src/components/ui/theme-toggle.tsx

```
@@ -290,41 +342,45 @@ export function ThemeToggle({
-                    themeIcons.light ?? ((props: { className?: string }) => <span {...props} />)
-                  const DarkIcon: IconComponent =
-                    themeIcons.dark ?? ((props: { className?: string }) => <span {...props} />)
-                  return (
-                    <>
-                      <LightIcon className="h-[1.2rem] w-[1.2rem] dark:hidden" />
-                      <DarkIcon className="h-[1.2rem] w-[1.2rem] hidden dark:block" />
-                    </>
-                  )
-                })()}
+            <span className="relative inline-block align-middle">
+              <style dangerouslySetInnerHTML={{ __html: themeIconCss }} />
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)